### DISCUSSION TITLE
Pass props to react-table ColumnDef

### DISCUSSION THREAD

**Discussion by kalaschnik** [TanStack/table#5334] [Q&A]
I struggle to pass a prop (specifically a URL searchParam) to my ColumnDef file (i.e., columns.tsx).

Visually that means:
![Untitled 3](https://github.com/TanStack/table/assets/9831678/3a191297-6651-4d8d-a92a-998927633119)

I've set-up a codesandbox [here](https://codesandbox.io/p/devbox/confident-babycat-k8c5yr).

**Comment by goatfryed** [comment]
Depending on your use case, you might want to either
- create the column definition dynamically with a factory/hook. In that case, you should wrap it in useMemo as well.
- use [table.meta](https://tanstack.com/table/v8/docs/api/core/table#meta) to pass extra props that can be accessed on lower levels

The following snippet might help you regarding usage of meta

```js
{
    accessorKey: "status",
    header: "Status",
    cell: ({ row, table }) => (
      <Link href={`http://example.com/${table.meta.owner}`}>{row.getValue("status")}</Link>
    ),
  },
```

**Reply by njoshi22** [reply]
Wow. Lifesaver --- did not know this existed and it just saved me a few hours of discovery!
